code/gamehelp/gameplayhelp.cpp

```diff
@@ -121,45 +131,306 @@ void gameplay_help_blit_control_line(int x, int y, int id)
-		strcat_s(buf, XSTR( ", ", 129));
+		strcat_s(buf, XSTR(", ", 129));
 		strcat_s(buf, ci->second.textify().c_str());
 	}
 
 	if (ci->empty()) {
-		strcpy_s(buf, XSTR( "no binding", 130));
+		strcpy_s(buf, XSTR("no binding", 130));
 	}
 
-	gr_string(x,y,buf,GR_RESIZE_MENU);
+	thisHelp.key.push_back(buf);
+	thisHelp.text.push_back(XSTR(ci->text.c_str(), CONTROL_CONFIG_XSTR + id));
+}
+
+void gameplay_help_init_control_line_raw(const char* control_text, const char* control_description, gameplay_help_section &thisHelp)
+{
+	thisHelp.key.push_back(control_text);
+	thisHelp.text.push_back(control_description);
+}
 
-//	gr_string(x+KEY_DESCRIPTION_OFFSET,y,ci->text,GR_RESIZE_MENU);
-	gr_string(x+KEY_DESCRIPTION_OFFSET, y, XSTR(ci->text.c_str(), CONTROL_CONFIG_XSTR + id), GR_RESIZE_MENU);
+// displays the given help key and text at the specified
+// x and y coordinates
+void gameplay_help_blit_help_text(int x, int y, const SCP_string *control_text, const SCP_string *control_description)
+{
+	gr_string(x, y, control_text->c_str(), GR_RESIZE_MENU);
+	gr_string(x + KEY_DESCRIPTION_OFFSET, y, control_description->c_str(), GR_RESIZE_MENU);
 }
 
-void gameplay_help_blit_control_line_raw(int x, int y, const char *control_text, const char *control_description)
+// displays a help section header at the specified y coordinate and an x coordinate
+// calculated from the header string's length
+void gameplay_help_blit_header_text(int y_offset, const SCP_string *header)
 {
-	gr_string(x,y,control_text,GR_RESIZE_MENU);
-	gr_string(x+KEY_DESCRIPTION_OFFSET,y,control_description,GR_RESIZE_MENU);
+	gr_set_color_fast(&Color_bright);
+	int w;
+	gr_get_string_size(&w, nullptr, header->c_str());
+	gr_printf_menu((gr_screen.clip_width - w) / 2, y_offset, "%s", header->c_str());
+	gr_set_color_fast(&Color_normal);
 }
 
 // game_play_help_set_title() will display the title for the help screen and
 // set the font for the rest of the screen
-void gameplay_help_set_title(const char *title)
+void gameplay_help_set_title(const SCP_string *title)
 {
 	int sy=TITLE_Y;
 	char buf[128];
 
 	gr_set_color_fast(&Color_bright);
 	int w;
-	gr_get_string_size(&w, NULL, title);
+	gr_get_string_size(&w, nullptr, title->c_str());
 
-	gr_printf_menu((gr_screen.clip_width_unscaled - w) / 2,sy,"%s", title);
+	gr_printf_menu((gr_screen.clip_width_unscaled - w) / 2,sy,"%s", title->c_str());
 
 	sprintf(buf, XSTR("Page %d of %d", 132), Current_help_page + 1, Gp_last_screen + 1);
 	gr_get_string_size(&w, NULL, buf);
 	gr_printf_menu((gr_screen.clip_width_unscaled - w) / 2, sy + gr_get_font_height() + 2, "%s", buf);
 	gr_set_color_fast(&Color_normal);
 }
 
+// Inits all the gameplay help text and returns it
+// in a nice organized vector of help sections
+SCP_vector<gameplay_help_section> gameplay_help_init_text()
+{
+	SCP_vector<gameplay_help_section> complete_help_text;
+
+	for (int i = 0; i <= GP_NUM_HELP_SECTIONS; i++) {
+
+		gameplay_help_section thisHelp;
+
+		switch (i) {
```

Comment:
This should not be necessary because adding sections here is done in the c++ only and with `default: continue;` it'll be easy for the person doing that work to not realize their new section isn't getting initialized. But sure. I've added it for your OCD.

Reply:
Default branches in switch statements is good programming practice, to defend against bugs both current and future.  And it doesn't *only* have to contain that.  It could also contain a logging message or a warning.

Reply:
> In situations where the switch statement can provably cover all possible cases, then the default case can be safely ignored

That's true as far as it goes, but if there's a chance that enums could be added to the list in the future, this assumption will be invalidated.  This is experience talking, not preference.

Reply:
Except when enums are added a case should also be added. There is no situation where you should have an enum without a case. Your experience is old and stale.

Reply:
"Should" is the key word.

Reply:
Yes, it is. You like the be the final gatekeeper. I dub you the one who gets to check for these cases being added in the future.